pyccel/ast/core.py

```diff
@@ -3801,23 +3806,25 @@ def __str__(self):
 
     def define_target(self, new_target):
         """
-        Add an additional target to the imports
+        Add an additional target to the imports.
+
+        Add an additional target to the imports.
         I.e. if imp is an Import defined as:
         >>> from numpy import ones
 
         and we call imp.define_target('cos')
         then it becomes:
         >>> from numpy import ones, cos
 
-        Parameter
-        ---------
-        new_target: str/AsName/iterable of str/AsName
-                    The new import target
+        Parameters
+        ----------
+        new_target : str | AsName | iterable[str | AsName]
+            The new import target.
         """
         if iterable(new_target):
-            self._target.update(new_target)
+            self._target.update({t: None for t in new_target})
         else:
-            self._target.add(new_target)
+            self._target[new_target] = None
```

Comment:
See above. Is `_format` needed?

Reply:
`_format` is only needed at the syntactic stage. I have added an assert for this a22b6b34

Reply:
Assertion failed. Format needed!